### PR TITLE
Enable squish decompression in runtime builds

### LINKED ISSUE #15983: Cannot decompress normal maps in WebGL 2.0
I think this is related to issue #15792

**Godot version:**
3.0 rc2

**OS/device including version:**
Ubuntu 17.10
Nvidia GTX 660 (using propietary driver version 384.111)
Tried on OSX sierra with same results.

**Issue description:**
Exporting to HTML5 renders without some textures. All textures are missing but the sky texture.

Issue #15792 suggests using the texture Format:
S 3tc On
Etc Off
Etc 2 On

But this justs makes it worse, I get a black screen and the console shows this error:
TypeError: Argument 9 of WebGL2RenderingContext.texImage2D is not an object.

I get the same results in firefox and chrome.

### PR DESCRIPTION
**Pull request by leonkrause**
Build squish decompression in no-tools builds. Allows decompressing RGTC when the relevant extension isn't available. Also add a check for the new WebGL extension for WebGL 2.
Fix #15983

### CODE REVIEW

**Comment by akien-mga** [comment]
Squish doesn't link properly in `tools=no` builds:
```
modules/libmodules.x11.opt.64.llvm.a(squish.x11.opt.64.llvm.o): In function `squish::CompressMasked(unsigned char const*, int, void*, int, float*)':
thirdparty/squish/squish.cpp:(.text+0x1a4): undefined reference to `squish::ColourSet::ColourSet(unsigned char const*, int, int)'
thirdparty/squish/squish.cpp:(.text+0x1c7): undefined reference to `squish::SingleColourFit::SingleColourFit(squish::ColourSet const*, int)'
thirdparty/squish/squish.cpp:(.text+0x1d2): undefined reference to `squish::ColourFit::Compress(void*)'
thirdparty/squish/squish.cpp:(.text+0x205): undefined reference to `squish::ClusterFit::ClusterFit(squish::ColourSet const*, int, float*)'
thirdparty/squish/squish.cpp:(.text+0x210): undefined reference to `squish::ColourFit::Compress(void*)'
thirdparty/squish/squish.cpp:(.text+0x22e): undefined reference to `squish::RangeFit::RangeFit(squish::ColourSet const*, int, float*)'
thirdparty/squish/squish.cpp:(.text+0x239): undefined reference to `squish::ColourFit::Compress(void*)'
thirdparty/squish/squish.cpp:(.text+0x243): undefined reference to `squish::ColourFit::~ColourFit()'
thirdparty/squish/squish.cpp:(.text+0x28a): undefined reference to `squish::ColourFit::~ColourFit()'
thirdparty/squish/squish.cpp:(.text+0x299): undefined reference to `squish::ColourFit::~ColourFit()'
thirdparty/squish/squish.cpp:(.text+0x2a8): undefined reference to `squish::ColourFit::~ColourFit()'
```

File: drivers/gles3/rasterizer_storage_gles3.cpp [hunk 1]
```diff
@@ -7673,7 +7673,7 @@ void RasterizerStorageGLES3::initialize() {
 	config.etc2_supported = true;
 	config.hdr_supported = false;
 	config.s3tc_supported = config.extensions.has("GL_EXT_texture_compression_dxt1") || config.extensions.has("GL_EXT_texture_compression_s3tc") || config.extensions.has("WEBGL_compressed_texture_s3tc");
-	config.rgtc_supported = config.extensions.has("GL_EXT_texture_compression_rgtc") || config.extensions.has("GL_ARB_texture_compression_rgtc");
+	config.rgtc_supported = config.extensions.has("GL_EXT_texture_compression_rgtc") || config.extensions.has("GL_ARB_texture_compression_rgtc") || config.extensions.has("EXT_texture_compression_rgtc") ;
```
Comment (akien-mga) [inline drivers/gles3/rasterizer_storage_gles3.cpp]: Space before `;` here.

**Comment by eska014** [comment]
Updated, the library is small, so no point taking efforts to build compression only for the editor. Emscripten does dead code elimination, so the size-critical platform benefits anyway

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.0.7.

**Pull request merged**